### ISSUE TITLE
[SIM110] Ruff enforces `all` instead of for loop inside async function.

### ISSUE THREAD

**Issue by real-yfprojects** [astral-sh/ruff#5098]
The following MRE violates the *SIM110* rule. 
```py
async def coroutine():
    ...


async def test():
    for i in range(10):
        if await coroutine():
            return True
    return False
```

Ruff suggests rewriting this as

```py
async def coroutine():
    ...


async def test():
    return any(await coroutine() for i in range(10))
```

However this code doesn't work since `any` doesn't support `AsyncGenerator`s. The same applies for `all` which is also covered by this rule.
Instead ruff should ignore for loops with `await` statements.

**Comment by charliermarsh** [comment]
Makes sense, thanks for reporting.

**Comment by real-yfprojects** [comment]
Thank you for fixing this @tjkuson
